pkg/core/interop/runtime/witness.go

```diff
@@ -10,7 +11,53 @@ import (
 
 // CheckHashedWitness checks given hash against current list of script hashes
 // for verifying in the interop context.
-func CheckHashedWitness(ic *interop.Context, hash util.Uint160) (bool, error) {
+func CheckHashedWitness(ic *interop.Context, v *vm.VM, hash util.Uint160) (bool, error) {
+	switch ic.Container.(type) {
+	case *transaction.Transaction:
+		tx := ic.Container.(*transaction.Transaction)
```

Comment:
`switch t := ic.Container.(type) {` so that you don't need to type cast in every `case`.
Or use `tx, ok := ` form of type cast, as there is only single case.